src/coprocessor/dag/batch/executors/simple_aggr_executor.rs

```diff
@@ -0,0 +1,882 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Simple aggregation is an aggregation that do not have `GROUP BY`s. It is more even more simpler
+//! than stream aggregation.
+
+use std::convert::TryFrom;
+use std::sync::Arc;
+
+use cop_datatype::{EvalType, FieldTypeAccessor};
+use tipb::executor::Aggregation;
+use tipb::expression::Expr;
+use tipb::expression::FieldType;
+
+use super::super::interface::*;
+use crate::coprocessor::codec::batch::LazyBatchColumnVec;
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::mysql::Tz;
+use crate::coprocessor::dag::aggr_fn::*;
+use crate::coprocessor::dag::exec_summary::ExecSummaryCollectorDisabled;
+use crate::coprocessor::dag::expr::{EvalConfig, EvalContext};
+use crate::coprocessor::dag::rpn_expr::types::RpnStackNode;
+use crate::coprocessor::dag::rpn_expr::RpnExpression;
+use crate::coprocessor::{Error, Result};
+
+// TODO: This macro may be able to be reused elsewhere.
+macro_rules! match_each_eval_type_and_call {
+    (match $target:expr => $fn_name:ident<T>($($parameters:expr),*)) => {{
+        match $target {
+            EvalType::Int => $fn_name::<Int>($($parameters),*),
+            EvalType::Real => $fn_name::<Real>($($parameters),*),
+            EvalType::Decimal => $fn_name::<Decimal>($($parameters),*),
+            EvalType::Bytes => $fn_name::<Bytes>($($parameters),*),
+            EvalType::DateTime => $fn_name::<DateTime>($($parameters),*),
+            EvalType::Duration => $fn_name::<Duration>($($parameters),*),
+            EvalType::Json => $fn_name::<Json>($($parameters),*),
+        }
+    }};
+}
+
+#[inline]
+// The `Box<>` wrapper cannot be removed actually. Thus disable the clippy rule. Additionally this
```

Comment:
Why can't the `Box` wrapper be removed? Is it ok to just call `aggr_fn_state.as_mut()` on calling?